xonsh/built_ins.py

```diff
@@ -351,8 +351,39 @@ def run_subproc(cmds, captured=True):
             stdin = PIPE
         else:
             stdin = prev_proc.stdout
-        proc = Popen(aliased_cmd, universal_newlines=uninew, env=ENV.detype(),
-                     stdin=stdin, stdout=stdout)
+        try:
+            proc = Popen(aliased_cmd, universal_newlines=uninew, env=ENV.detype(),
+                         stdin=stdin, stdout=stdout)
+        except FileNotFoundError:
+            cmd = aliased_cmd[0]
+            print('xonsh: subprocess mode: command not found: {0}'.format(cmd))
```

Comment:
Would you mind moving most of this block to its own function and the calling it from here? This function is already getting kinda long :)